synthesizer/process/src/stack/register_types/initialize.rs

```diff
@@ -186,7 +186,7 @@ impl<N: Network> RegisterTypes<N> {
                 for operand in async_.operands() {
                     if let Operand::Register(register) = operand {
                         if let Ok(RegisterType::Future(locator)) = register_types.get_type(stack, register) {
-                            assert!(future_registers.remove(&(register.clone(), locator)));
+                            assert!(future_registers.swap_remove(&(register.clone(), locator)));
```

Comment:
we do want `swap_remove` here, because right after this loop we `ensure` that the `future_registers` is empty, meaning the order is no longer significant

Reply:
@d0cd @evan-schott  can you confirm `swap_remove` is the intended behavior? (i understand it was technically swap_remove, want to ensure it truly shouldn't be shift_remove)

Reply:
Confirmed that `swap_remove` is intended since we do not rely on the order of `future_registers` after the removes are done.